src/Test/QuickCheck/Generic.purs

```diff
@@ -0,0 +1,64 @@
+-- | Generic deriving for `Arbitrary` and `Coarbitrary` instances.
+
+module Test.QuickCheck.Generic 
+  ( gArbitrary
+  , gCoarbitrary
+  ) where
+
+import Prelude
+
+import Data.Maybe (Maybe(..))
+import Data.Maybe.Unsafe (fromJust)
+import Data.Array (uncons)
+import Data.Traversable (traverse)
+import Data.Generic
+import Data.Monoid.Endo
+import Data.Foldable (Foldable, foldMap)
+
+import Control.Lazy (defer)
+
+import Test.QuickCheck.Gen
+import Test.QuickCheck.Arbitrary
+
+-- | Generate arbitrary values for any `Generic` data structure.
+-- |
+-- | _Note_: this function _is_ partial, since it does not handle ADTs with zero constructors
+-- | such as `Void`.
```

Comment:
So imagine if one had a type `Either Void Int` and tried to use generic deriving on it -- the concern I think is that you might hit the `Left` case and fail instead of just _only_ taking the right path?